### PR TITLE
Avoid most trimming and preserve comment structure.

### PR DESCRIPTION
**Pull request by 0ndorio**
This fixes two additional documentation related issues: #184 & #374. Both effect nearly the same parts of the source code. It also updates the `documentation` test to verify the new expected behavior.

---

Feel free to nitpick. I will try to come up with all required changes to get this merged.

### CODE REVIEW

File: src/bindgen/ir/annotation.rs [hunk 1]
```diff
@@ -52,8 +52,15 @@ impl AnnotationSet {
     pub fn load(attrs: &[syn::Attribute]) -> Result<AnnotationSet, String> {
         let lines: Vec<String> = attrs
             .get_comment_lines()
-            .into_iter()
-            .filter(|x| !x.is_empty() && x.starts_with("cbindgen:"))
+            .iter()
+            .filter_map(|line| {
+                let line = line.trim_start();
+                if line.is_empty() || !line.starts_with("cbindgen:") {
```
Comment (emilio) [inline src/bindgen/ir/annotation.rs]: I think we could avoid a bunch of extra allocations here...

Any reason we can't do:

```
let lines = attrs.get_comment_lines();
let lines: Vec<&str> = lines.iter().filter_map(..).collect();
```

Also, the `is_empty()` check looks redundant to me (was redundant already, but I can't see how an empty line can start with `"cbindgen:"` ;))

File: src/bindgen/ir/annotation.rs [hunk 2]
```diff
@@ -62,12 +69,7 @@ impl AnnotationSet {
 
         // Look at each line for an annotation
         for line in lines {
```
Comment (emilio) [inline src/bindgen/ir/annotation.rs]: `debug_assert!(line.starts_with("cbindgen:"))` maybe?

Comment (0ndorio) [reply src/bindgen/ir/annotation.rs]: Sounds like a good idea.

File: src/bindgen/ir/annotation.rs (hunk 1 above)
Comment (0ndorio) [reply src/bindgen/ir/annotation.rs]: Absolutely no reason! Going to change this asap.

Comment (0ndorio) [reply src/bindgen/ir/annotation.rs]: Included via rebase.

File: src/bindgen/ir/annotation.rs (hunk 2 above)
Comment (0ndorio) [reply src/bindgen/ir/annotation.rs]: Included via rebase

**Pull request merged**